parser.go

```diff
@@ -157,6 +157,25 @@ func (p *Parser) Progress() float32 {
 	return float32(p.currentFrame) / float32(p.header.PlaybackFrames)
 }
 
+// Weapons returns all weapons currently in the game.
+func (p *Parser) WeaponsByEntityID() map[int]*common.Equipment {
+	res := make(map[int]*common.Equipment)
+	for k, v := range p.weapons {
+		res[k] = v
+	}
+
+	return res
+}
+
+// WeaponTypeByEntityID returns the weapon type by its entity-ID.
+func (p *Parser) WeaponTypeByEntityID(entityID int) common.EquipmentElement {
```

Comment:
As I mentioned a bit early, if we will go with the desing of all other entity getters inside `GameState`, this would be indeed redundant.

Anyhow, I just didn't think much of how it would look in terms of going in line with other API, I just added this after realizing that this is actually all the needed functionality and that I should have probably gone with this proposition initially instead of making all the weapons visible to the user but looks like it's a bit late for going back 😃 Now I see how it really doesn't fit there so yeah probably my little obsession with no-copy behaviour shouldn't be praised and it is better to make a more user-friendly interface.

Reply:
I think it's good that we expose the whole data, could be useful to someone. see above comment about leaving the copy-stuff out.

Reply:
Removed the method